File: HaberUygulamaJetpack/app/src/main/java/com/example/haberuygulamajetpack/viewmodel/HaberDetayViewModel.kt
package com.example.haberuygulamajetpack.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.haberuygulamajetpack.deo.HaberDao
import com.example.haberuygulamajetpack.model.HaberModel
import com.example.haberuygulamajetpack.model.YorumInsertRequest

import com.example.haberuygulamajetpack.model.YorumModel
import com.example.haberuygulamajetpack.servis.ApiClient
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch


// HaberDetayViewModel - Parametre ile başlatılıyor
class HaberDetayViewModel(private val haberDao: HaberDao = HaberDao(ApiClient.retrofit)) : ViewModel() {
    private val _yorumlar = MutableStateFlow<List<YorumModel>>(emptyList())
    val yorumlar: StateFlow<List<YorumModel>> = _yorumlar

    // Yorumları yükle
    fun loadYorumlar(haberId: Int) {
        viewModelScope.launch {
            _yorumlar.value = haberDao.getYorumlar(haberId) ?: emptyList()
        }
    }

    // Yorum ekle
    fun yorumEkle(haberId: Int, takmaAd: String, yorumMetni: String) {
        viewModelScope.launch {
            val response = haberDao.insertYorum(
                YorumInsertRequest(haberId, takmaAd, yorumMetni)
            )
            if (response?.success == true) {
                loadYorumlar(haberId)
            }
        }
    }
}

File: flutter_haber_uygulama/lib/viewmodel/haberdetay_viewmodel.dart
// viewmodel/haber_detay_viewmodel.dart
import 'package:flutter/foundation.dart';

import '../dao/haber_dao.dart';
import '../model/yorum_insert_request.dart';
import '../model/yorum_model.dart';

class HaberDetayViewModel extends ChangeNotifier {
  final HaberDao haberDao;

  HaberDetayViewModel({HaberDao? dao}) : haberDao = dao ?? HaberDao();

  // Kotlin: private val _yorumlar = MutableStateFlow<List<YorumModel>>(emptyList())
  List<YorumModel> yorumlar = [];

  bool loading = false;
  String? error;

  // Kotlin: fun loadYorumlar(haberId: Int)
  Future<void> loadYorumlar(int haberId) async {
    loading = true;
    error = null;
    notifyListeners();

    try {
      yorumlar = await haberDao.getYorumlar(haberId) ?? [];
    } catch (e) {
      error = e.toString();
      yorumlar = [];
    } finally {
      loading = false;
      notifyListeners();
    }
  }

  // Kotlin: fun yorumEkle(haberId: Int, takmaAd: String, yorumMetni: String)
  Future<void> yorumEkle(int haberId, String takmaAd, String yorumMetni) async {
    loading = true;
    error = null;
    notifyListeners();

    try {
      final resp = await haberDao.insertYorum(
        YorumInsertRequest(
          haber_id: haberId,
          takma_ad: takmaAd,
          yorum_metni: yorumMetni,
        ),
      );

      if (resp?.success == true) {
        await loadYorumlar(haberId);
      } else {
        error = resp?.error ?? 'Yorum eklenemedi';
      }
    } catch (e) {
      error = e.toString();
    } finally {
      loading = false;
      notifyListeners();
    }
  }
}
